core/modules/menu/menu.admin.inc

```diff
@@ -412,8 +412,38 @@ function menu_edit_item($form, &$form_state, $type, $item, $menu) {
     '#description' => t('When displayed in a "hierarchical tree" menu style, children of expanded menu links will always be visible.'),
   );
 
-  // Generate a list of possible parents (not including this link or descendants).
-  $options = menu_parent_options(menu_get_menus(), $item);
+  $separate_menu = config_get('menu.settings', 'menu_separate_menu_selector');
+  $menu_name = $item['menu_name'];
+  $menus = menu_get_menus();
+  if ($separate_menu) {
+    $form['parent_menu'] = array(
+      '#type' => 'select',
+      '#title' => t('Menu'),
+      '#default_value' => $menu_name,
+      '#options' => $menus,
+      '#description' => t('Select the menu that this menu link should belong to.'),
+      '#attributes' => array('class' => array('parent-menu-select')),
+      '#ajax' => array(
+        'callback' => 'menu_parent_menu_callback',
+      ),
+    );
+    if (isset($form_state['values']['parent_menu'])) {
+      $menu_name = $form_state['values']['parent_menu'];
+    }
+    // Generate a list of possible parents (not including this link or
+    // descendants) from the selected menu.
+    $options = menu_parent_options(array($menu_name => $menus[$menu_name]), $item);
+    foreach ($options as $key => &$option) {
+      if (strpos($key, ':0') == strlen($key) - 2) {
+        $option .= ' ' . t('(top level)');
+      }
+    }
```

Comment:
Why not simply append the text to the first menu item here? Isn't the menu itself always the 1st option?

Reply:
Yes, looks like it is. Updated (thanks!).